packages/wonder-blocks-tooltip/util/visibility-modifier.js

```diff
@@ -0,0 +1,70 @@
+// @flow
+/**
+ * A modifier for use with popper.js or react-popper.
+ * This looks at the scroll parents of the reference element to determine
+ * overall visibility.
+ */
+import {default as PopperJS} from "popper.js";
+
+import {getElementIntersection} from "@khanacademy/wonder-blocks-core";
+
+import isObscured from "./is-obscured.js";
+
+/**
+ * The function that implements the modifier.
+ */
+function visibilityModifierFn(data: any) {
+    const anchorElement = data.instance.reference;
+
+    // First, we see how the element intersects with its scroll parents.
+    // If it doesn't, then we should hide it.
+    // Otherwise, we check to see if anything else obscures the component (like
+    // a fixed or absolute positioned element).
+    const {horizontal, vertical} = getElementIntersection(anchorElement);
+    const hide =
+        horizontal !== "within" ||
+        vertical !== "within" ||
+        isObscured(anchorElement);
+
+    // If we're hidden, we mimic what the built-in hide method does,
+    // and set the hide flag and the OOB attribute with appropriate
+    // short-circuiting.
+    if (hide) {
+        if (data.hide) {
+            return data;
+        }
+        data.hide = true;
+        data.attributes["x-out-of-boundaries"] = "";
+    } else {
+        // Avoid unnecessary DOM access if visibility hasn't changed
+        if (!data.hide) {
+            return data;
+        }
+        data.hide = false;
+        data.attributes["x-out-of-boundaries"] = false;
+    }
+
+    // Always have to return the data object to ensure the modifier chain
+    // in popper.js is unbroken.
+    return data;
```

Comment:
Maybe link to the docs for popper.js modifiers in the docstring at the beginning fo this file.

Reply:
Good idea.

Reply:
https://popper.js.org/popper-documentation.html#modifiers